pkg/ingestor/collector/collector.go

```diff
@@ -15,4 +15,77 @@
 
 package collector
 
-type Collector interface{}
+import (
+	"context"
+
+	"github.com/guacsec/guac/pkg/ingestor/processor"
+	"github.com/sirupsen/logrus"
+)
+
+type Collector interface {
+	// Is the collector enabled
+	Enabled(ctx context.Context) bool
+	// Retrieve the artifacts from the collector
+	RetrieveArtifacts(ctx context.Context) ([]*processor.Document, error)
+	// Indicated when the collector is done
+	IsDone() bool
+	// Type of the collector
+	Type() CollectorType
+}
+
+type CollectorType string
+
+const (
+	CollectorGCS    = "GCS"
+	CollectorOCI    = "OCI"
+	CollectorPubSub = "PubSub"
+	CollectorRekor  = "Rekor"
+)
```

Comment:
nit: I'm thinking we can keep these private to each collector type instead of here. This way, when we want to add support for a new collector, we'd only add files, instead of also changing here.

I don't know if go supports plugins loaded at runtime, but if it does, having each collector type be isolated in its own file would enable not recompiling the entire thing when we want to add support for a new collector

Reply:
done.